### PR TITLE
f-offers@v0.3.0 - Adding search box component

### PR DESCRIPTION
**Pull request by matthewhardern**
This PR adds a search box component that implements the `f-searchbox` component. I have added unit tests to support this new component.

- [x] Unit tests have been [created|updated]

### CODE REVIEW

File: packages/components/organisms/f-offers/src/components/SearchBox.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+} from '../store/types';
+import { POSTCODE_EMPTY, POSTCODE_INVALID } from '../constants';
+
+export default {
+    name: 'OffersSearchBox',
+
+    components: {
+        SearchBox
+    },
+
```
Comment (minrwhite) [inline packages/components/organisms/f-offers/src/components/SearchBox.vue:40]: Worth registering the offersHeaderSearch module here within a beforeCreate [similarly to how it's done in `f-searchbox`](https://github.com/justeat/fozzie-components/blob/master/packages/components/molecules/f-searchbox/src/components/Form.vue#L280-L284)?

Comment (matthewhardern) [reply packages/components/organisms/f-offers/src/components/SearchBox.vue:40]: yes your right my bad, thanks

**Review by minrwhite — DISMISSED** [review]

File: packages/components/organisms/f-offers/src/store/types.js [hunk 2]
```diff
@@ -6,3 +6,5 @@ export const ACTION_VALID_SAVED_ADDRESS_SEARCH = 'validSavedAddressSearch';
 export const ACTION_VALID_SEARCH = 'validSearch';
 
 export const MUTATION_EMIT_SEARCH_START = 'searchbox-focus';
+
+export const OFFERS_SEARCH_VUEX_MODULE_NAMESPACE = 'offersSearch';
```
Comment (minrwhite) [inline packages/components/organisms/f-offers/src/store/types.js]: I'd prefer if the namespace for this was at the beginning to match the other parts of this file, e.g:
```suggestion
export const VUEX_MODULE_NAMESPACE_OFFERS_SEARCH = 'offersSearch';
```

**Review by minrwhite — DISMISSED** [review]

File: packages/components/organisms/f-offers/src/store/types.js (hunk 2 above)
Comment (matthewhardern) [reply packages/components/organisms/f-offers/src/store/types.js]: agreed good idea

**Review by minrwhite — APPROVED** [review]

**Review by charliearlie — APPROVED** [review]

**Pull request merged**